File: responses/BWJ4RW-5Q8/3_longest_common_prefix.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
  return "" if a.empty?
  prefix = ''
  a[0].split('').each do |c|
      prefix += c
      unless a.map{ |w| w.include? prefix }.all?
        return prefix[0...-1]
      end
  end
end

File: responses/Q3A2XR-3VW/3_longest_common_prefix.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    min_len = min([len(a) for a in A])
    prefix_len = -1
    for j in range(min_len):
        prefix = A[0][:j+1]
        for i in range(1, len(A)):
            if A[i][:j+1] != prefix:
                prefix_len = j
                break
        if prefix_len >= 0:
            break
    return A[0][:prefix_len]
